core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -237,8 +211,11 @@ public void releaseWorkerClient(BlockWorkerClient blockWorkerClient) {
     // close the client.
     if (blockWorkerClient.isLocal()) {
       // Return local worker client to its resource pool.
-      Preconditions.checkState(mLocalBlockWorkerClientPool != null);
-      mLocalBlockWorkerClientPool.release(blockWorkerClient);
+      WorkerNetAddress address = blockWorkerClient.getWorkerNetAddress();
+      synchronized (mLocalBlockWorkerClientPoolMap) {
+        Preconditions.checkState(mLocalBlockWorkerClientPoolMap.containsKey(address));
```

Comment:
If we no longer have a pool to return it to (only occurs due to `reset()` being called?), should we just treat it like a remote worker client and close it? Crashing seems a bit extreme?